packages/next/src/export/index.ts

```diff
@@ -733,6 +776,15 @@ export default async function exportApp(
           )
 
           const orig = join(distPagesDir, route)
+          const handlerSrc = `${orig}.body`
+          const handlerDest = join(outDir, route)
+
+          if (isAppRouteHandler && (await exists(handlerSrc))) {
+            await promises.mkdir(dirname(handlerDest), { recursive: true })
+            await promises.copyFile(handlerSrc, handlerDest)
```

Comment:
We might need to error here if custom headers are set in the `.meta` file or a custom status code as those won't work with `next export`

Reply:
Good point, I created NEXT-823 to track this 👍